--- src/PureScript/Backend/Optimizer/Semantics.purs
+++ src/PureScript/Backend/Optimizer/Semantics.purs
@@ -1249,6 +1249,15 @@
           Just (InlineArity n) | Array.length args2 >= n ->
             Just $ evalSpine env (eval (envForGroup env ref (InlineProp prop) group) expr) spine
           _ ->
+            Nothing
+      _ ->
+        Nothing
+  [ ExternApp _, ExternAccessor (GetCtorField _ _ _ _ _ _), ExternAccessor (GetProp prop) ] ->
+    case impl of
+      ExternExpr group fn -> do
+        let ref = EvalExtern qual
+        case Map.lookup ref e.directives >>= Map.lookup (InlineSpineProp prop) of
+          Just InlineNever ->
+            Just $ neutralSpine (NeutStop qual) spine
+          Just InlineAlways ->
+            Just $ evalSpine env (eval (envForGroup env ref (InlineSpineProp prop) group) fn) spine
+          _ ->
             Nothing
       _ ->
         Nothing
