--- src/PureScript/Backend/Optimizer/Monomorphize.purs
+++ src/PureScript/Backend/Optimizer/Monomorphize.purs
@@ -586,7 +586,7 @@
                 ( \(Tuple specKey info) ->
                     let 
                       hasTV = hasTypeVariables info.instType
-                      _ = if qualName == "Control.Semigroupoid.compose" then unsafePerformEffect (Console.log ("compose specKey in typeMap: " <> specKey) *> pure unit) else unit
+                      _ = if qualName == "Control.Semigroupoid.compose" && hashString specKey == "2467103119" then unsafePerformEffect (Console.log ("Checking hasTypeVariables for compose__2467103119: " <> show hasTV) *> pure unit) else unit
                       forceUse = if hasTV then true else false
                     in
                       if forceUse then Nothing
