--- src/PureScript/Backend/Optimizer/Monomorphize.purs
+++ src/PureScript/Backend/Optimizer/Monomorphize.purs
@@ -403,6 +403,11 @@
                             exprWithDicts = applyDicts info.dictArgs expr
                             resolvedExpr = resolveGlobals definerMod Set.empty exprWithDicts
                             specializedExpr = rewriteExpr globalAstMap Map.empty substFn (monomorphizeExpr modNameStr instMap Map.empty resolvedExpr)
+                            _ = if modNameStr == "Data.Monoid" && unwrap id == "mempty" && String.contains (Pattern "193435443") (hashString (mangleType (defaultToAny info.instType))) then
+                                  unsafePerformEffect $ do
+                                    log ("MEMPTY ETA EXPANDED 193435443 BEFORE REWRITE: " <> show exprWithDicts)
+                                    log ("MEMPTY ETA EXPANDED 193435443 AFTER REWRITE: " <> show specializedExpr)
+                                else unit
                             etaExpandedExpr = case specializedExpr of
                               ExprAbs _ _ _ -> specializedExpr
                               _ ->
