--- src/PureScript/Backend/Optimizer/Convert.purs
+++ src/PureScript/Backend/Optimizer/Convert.purs
@@ -287,7 +287,7 @@
   let evalEnv = Env { currentModule: env.currentModule, evalExternRef: makeExternEvalRef env, evalExternSpine: makeExternEvalSpine env, locals: [], directives: env.directives }
   let qualifiedIdent = Qualified (Just env.currentModule) ident
   let backendExpr = toBackendExpr cfn env
-  let _ = if ident == Ident "never" then unsafePerformEffect (Effect.Class.Console.log ("\n--- NEVER BEFORE BACKEND ---\n" <> Debug.show cfn <> "\n--- NEVER BACKEND EXPR ---\n" <> printExpr backendExpr)) else unit
+  let _ = if ident == Ident "never" then unsafePerformEffect (Effect.Class.Console.log ("\n--- NEVER BACKEND EXPR ---\n" <> printExpr backendExpr)) else unit
   let enableTracing = Set.member qualifiedIdent env.traceIdents
   let
     mbType = case backendExpr of
