--- src/PureScript/Backend/Optimizer/Convert.purs
+++ src/PureScript/Backend/Optimizer/Convert.purs
@@ -380,21 +380,24 @@
         Nothing
 
 isTypeClassDictionaryWithProps :: Expr Ann -> Tuple Boolean (Array String)
-isTypeClassDictionaryWithProps = case _ of
-  ExprAbs (Ann { type: Just ty }) _ body | isConstrainedType ty ->
-    isTypeClassDictionaryWithProps body
-  ExprAbs _ _ body -> isTypeClassDictionaryWithProps body
-  ExprLet _ _ body -> isTypeClassDictionaryWithProps body
-  ExprTypeApp _ expr' _ -> isTypeClassDictionaryWithProps expr'
-  ExprApp _ lhs (ExprLit _ (LitRecord props))
-    | Just meta <- getConstructorMeta lhs ->
-        if meta == IsTypeClassConstructor || meta == IsNewtype then Tuple true (map propKey props) else Tuple false []
-  expr -> case unrollApps expr of
-    Tuple (ExprVar (Ann { meta: Just meta }) _) _
-      | meta == IsTypeClassConstructor || meta == IsNewtype -> Tuple true []
-    _ -> Tuple false []
+isTypeClassDictionaryWithProps = go false
   where
+  go isClass = case _ of
+    ExprAbs (Ann { type: Just ty }) _ body | isConstrainedType ty ->
+      go true body
+    ExprAbs _ _ body -> go isClass body
+    ExprLet _ _ body -> go isClass body
+    ExprTypeApp _ expr' _ -> go isClass body
+    ExprApp _ lhs (ExprLit _ (LitRecord props))
+      | Just meta <- getConstructorMeta lhs ->
+          if meta == IsTypeClassConstructor || meta == IsNewtype then Tuple true (map propKey props) else Tuple false []
+    ExprLit _ (LitRecord props) | isClass ->
+      Tuple true (map propKey props)
+    expr -> case unrollApps expr of
+      Tuple (ExprVar (Ann { meta: Just meta }) _) _
+        | meta == IsTypeClassConstructor || meta == IsNewtype -> Tuple true []
+      _ -> Tuple false []
+
   unrollApps :: Expr Ann -> Tuple (Expr Ann) (Array (Expr Ann))
   unrollApps = go []
     where
