--- src/PureScript/Backend/Optimizer/Monomorphize.purs
+++ src/PureScript/Backend/Optimizer/Monomorphize.purs
@@ -237,6 +237,7 @@
              stripForAlls = case _ of
                ForAll _ b -> stripForAlls b
                x -> x
              instType = stripTypeVariables (substituteExprType subst (stripForAlls genericType))
+             _ = if String.contains (Pattern "filter") qualName then trace ("\n\n[TRACE 1.6.3.3] collectExpr ExprApp for " <> qualName <> "\n  hasTypeVars instType: " <> show (hasTypeVariables instType)) \_ -> unit else unit
              { dictArgs, normalArgs } = partitionArgs genericType args
              staticNormalArgs = Array.filter isStatic normalArgs
