--- src/PureScript/Backend/Optimizer/Convert.purs
+++ src/PureScript/Backend/Optimizer/Convert.purs
@@ -362,8 +362,8 @@
       , dictSize <= 512000 ->
           Just $ Map.singleton InlineRef InlineAlways
     cfn' | Tuple isDict props <- isTypeClassDictionaryWithProps cfn', isDict ->
-      let _ = Debug.trace ("Found dict: " <> show dictSize) \_ -> unit in
-      if dictSize <= 512000 then
+      let _ = Debug.trace ("Found dict with props! Size: " <> show dictSize) \_ -> unit in
+      if dictSize <= 512000000 then
         Just $ Map.fromFoldable $
           [ Tuple InlineRef InlineAlways ] <> (props >>= \p -> [ Tuple (InlineProp p) InlineAlways, Tuple (InlineSpineProp p) InlineAlways ])
       else Nothing
