--- src/PureScript/Backend/Optimizer/Monomorphize.purs
+++ src/PureScript/Backend/Optimizer/Monomorphize.purs
@@ -875,6 +875,7 @@
                       unifySpine _ _ s = s
                       
                       substType = unifySpine genericType args substFromTypeArgs
+                      _ = unsafePerformEffect (Console.log ("collectLocalExpr substType=" <> show (Map.keys substType)) *> pure unit)
                     in
                       if not (Map.isEmpty substType) then
                         let
