--- src/PureScript/Backend/Optimizer/Convert.purs
+++ src/PureScript/Backend/Optimizer/Convert.purs
@@ -363,6 +363,7 @@
           Just $ Map.singleton InlineRef InlineAlways
     cfn' | Tuple isDict props <- isTypeClassDictionaryWithProps cfn', isDict
          , dictSize <= 512000 ->
+      let _ = Debug.trace ("Found dict: " <> show dictSize) \_ -> unit in
       Just $ Map.fromFoldable $
         [ Tuple InlineRef InlineAlways ] <> (props >>= \p -> [ Tuple (InlineProp p) InlineAlways, Tuple (InlineSpineProp p) InlineAlways ])
     _ -> case backendExpr of
